shotover-proxy/src/transforms/protect/crypto.rs

```diff
@@ -13,7 +14,7 @@ use sqlparser::ast::Value as SQLValue;
 #[derive(Serialize, Deserialize)]
 struct Protected {
     cipher: Vec<u8>,
-    nonce: Nonce,
+    nonce: Vec<u8>,
```

Comment:
ughh, really unfortunate the chacha crate was written before const generics, not sure how feasibile it is to just go and change something like that in a crypto crate under a lot of scrutiny.

That said we can still make Nonce and friends serializable by adding the following to our cargo.toml
`generic-array = { version = "0.14", features = ["serde"] }`
generic-array is what the chacha crate is using instead of const generics.

Reply:
I'll raise an issue on the repo for it but that's an alright workaround.

Reply:
Alright, there's an issue on the repo for it which I've subscribed to. If it gets any updates I'll fix it in our codebase.